screens/Announcements.js

```diff
@@ -0,0 +1,257 @@
+})
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(Announcements)
+
+class Row extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this._active = new Animated.Value(0)
+
+    this._style = {
+      ...Platform.select({
+        ios: {
+          transform: [
+            {
+              scale: this._active.interpolate({
+                inputRange: [0, 1],
+                outputRange: [1, 1.1]
+              })
+            }
+          ],
+          shadowRadius: this._active.interpolate({
+            inputRange: [0, 1],
+            outputRange: [2, 10]
+          })
+        },
+
+        android: {
+          transform: [
+            {
+              scale: this._active.interpolate({
+                inputRange: [0, 1],
+                outputRange: [1, 1.07]
+              })
+            }
+          ],
+          elevation: this._active.interpolate({
+            inputRange: [0, 1],
+            outputRange: [2, 6]
+          })
+        }
+      })
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.active !== nextProps.active) {
+      Animated.timing(this._active, {
+        duration: 300,
+        easing: Easing.bounce,
+        toValue: Number(nextProps.active)
+      }).start()
+    }
+  }
+
+  render() {
+    const { data } = this.props
+
+    return (
+      <Animated.View>
+        <View key={data.id}>
+          <View
+            style={[styles.flex_dir_row, { paddingTop: 4, marginBottom: 17 }]}
+          >
+            <View style={{ width: '45%' }}>
+              <Text style={{ fontSize: 15 }}>{data.title}</Text>
+            </View>
+
+            <View style={[{ width: '55%' }]}>
+              <Icon
+                name="md-create"
+                size={25}
+                color="#f18d1a"
+                onPress={() =>
+                  this.props.navigation.navigate('AnnouncementsEdit', {
+                    announcementId: data.id,
+                    initialValues: data
+                  })
+                }
+                style={{ textAlign: 'right', marginRight: 30 }}
+              />
+              <IonIcon
+                name={data.titleIcon}
+                size={28}
+                color="#f18d1a"
+                //onPress={() => fields.push()}
+                style={{ position: 'absolute', right: 0 }}
+              />
+            </View>
+          </View>
+
+          <View
+            style={[
+              { backgroundColor: '#f1f1f1', padding: 20 },
+              styles.mgrbtn20
+            ]}
+          >
+            <Markdown style={styles.markDownStyle}>
+              ### {data.markdownContent}
+              {'\n'}
+            </Markdown>
+            <Text
+              style={{
+                textAlign: 'center',
+                marginTop: 20,
+                fontSize: 15,
+                marginBottom: -10
+              }}
+            >
+              Order&nbsp;{data.order}
```

Comment:
I think I may have an approach for changing the order. 

Let's say we have 3 announcements (0, 1, 2), when you move announcement 0 to behind announcement 1, then announcement 0's order becomes 1, and announcement 1's order becomes (order - 1), which is 0. 

When you move announcement 0 to behind announcement 2, then announcement 0's order becomes 2, and announcement 2's order becomes (order - 1), which is 1.

So, when you move an announcement, you will make two POST /announcements API to change the order of two announcements.

Reply:
for 5 annocemnts(0,1,2,3,4,5), when move announcements 0 to 5, then annoc_0's index become 5, annoc_5's index become 4, then how others annocs' index automatically updated ? is it enough two POST.